pkg/ssh/ssh.go

```diff
@@ -0,0 +1,71 @@
+package ssh
+
+import (
+	"fmt"
+
+	"golang.org/x/crypto/ssh"
+)
+
+func Create(options *ConnectionCreateOptions, kind EngineMode) error {
+	var err error
+	if kind == NativeMode {
+		_, err = nativeConnectionCreate(*options)
+		if err != nil {
+			return err
+		}
+	} else {
+		_, err = golangConnectionCreate(*options)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Dial(options *ConnectionDialOptions, kind EngineMode) (*ssh.Client, error) {
+	var rep *ConnectionDialReport
+	var err error
+	if kind == NativeMode {
+		return nil, fmt.Errorf("ssh dial failed: you cannot create a dial-able client with native ssh")
+	}
+	rep, err = golangConnectionDial(*options)
+	if err != nil {
+		return nil, err
+	}
+	return rep.Client, nil
+}
+
+func Exec(options *ConnectionExecOptions, kind EngineMode) (string, error) {
+	var rep *ConnectionExecReport
+	var err error
+	if kind == NativeMode {
+		rep, err = nativeConnectionExec(*options)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		rep, err = golangConnectionExec(*options)
+		if err != nil {
+			return "", err
+		}
+	}
+	return rep.Response, nil
+}
+
+func Scp(options *ConnectionScpOptions, kind EngineMode) (string, error) {
+	var rep *ConnectionScpReport
+	var err error
+	if kind == NativeMode {
+		rep, err = nativeConnectionScp(*options)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		rep, err = golangConnectionScp(*options)
+		if err != nil {
```

Comment:
Move this block to outside of if, and remove if err!=nil below.

Reply:
if I take away the err != nil i'll seg fault because report is nil